suite/tests/CMakeLists.txt

```diff
@@ -4073,6 +4073,17 @@ if (UNIX)
     "-enable_reset -reset_at_fragment_count 100" "")
   torunonly(linux.clone-reset linux.clone linux/clone.c
     "-enable_reset -reset_at_fragment_count 100" "")
+  if (AARCHXX)
+    # Test our diagnostic option -steal_reg_at_reset, which is also a stress
+    # test of reset and register stealing.
+    if (ARM)
+      set(new_reg 9)
+    else ()
+      set(new_reg 29)
+    endif ()
+    torunonly(linux.steal-switch linux.thread linux/thread.c
+      "-enable_reset -reset_at_fragment_count 100 -steal_reg_at_reset ${new_reg}" "")
```

Comment:
This test may have the same issue like `linux.thread-reset` and `linux.clone-reset`, with triggering reset at the right time. A reset event will execute `translate_from_synchall_to_dispatch` (therefore the new code added in `core/synch.c`) only if >1 threads are active. So, the `n` in `-reset_at_fragment_count n` needs to be hand-tuned. I'm trying to fix this issue in these tests in #4498 and #4496).

Does the test fail without the stolen reg store in core/synch.c?

Reply:
For my purposes I don't care about the synchall path: I just want to replace the entire code cache and generated code and stubs that were using r28 with new code using r29.  Any fragment count will do.  If the test were to be improved, it would be to add app code that checks all register values, but I don't think it's worth the effort for a sanity check test that just makes sure the base option continues to work.